storage/mutable_segment.go

```diff
@@ -281,6 +357,61 @@ func (s *mutableSeg) Close() {
 	s.fields = nil
 }
 
+func (s *mutableSeg) collectFieldsForRawQueryWithLock(
+	q query.ParsedRawQuery,
+) (
+	fieldTypes []field.ValueTypeSet,
+	fieldIndexMap []int,
+	queryFields []indexfield.DocsField,
+	err error,
+) {
+	numFieldsForQuery := q.NumFieldsForQuery()
+	fieldTypes = make([]field.ValueTypeSet, numFieldsForQuery)
+	fieldIndexMap = make([]int, numFieldsForQuery)
+	queryFields = make([]indexfield.DocsField, 0, numFieldsForQuery)
+	allAllowedFieldTypes := q.AllowedFieldTypes
```

Comment:
this rename doesn't seem super necessary

Reply:
np, can remove.